src/Handlebars/Context.php

```diff
@@ -32,21 +37,42 @@ class Context
      */
     protected $index = [];
 
+    /**
+     * @var array dataStack stack for data within sections
+     */
+    protected $dataStack = [];
+
     /**
      * @var array key stack for objects
      */
     protected $key = [];
 
+    /**
+     * @var bool enableDataVariables true if @data variables should be used.
+     */
+    protected $enableDataVariables = false;
+
     /**
      * Mustache rendering Context constructor.
      *
      * @param mixed $context Default rendering context (default: null)
+     * @param array $options Options for the context. It may contain the following: (default: empty array)
+     *                       enableDataVariables => Boolean, Enables @data variables (default: false)
```

Comment:
Should include an @throws annotation indicating that an InvalidArgumentException can be thrown.